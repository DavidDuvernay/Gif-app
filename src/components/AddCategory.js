import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories, categories } ) => {
 
    const [inputValue, setInputValue] = useState('');

    const handleinputChange = ( e ) => {
        
        setInputValue( e.target.value )
    };

    const handleSubmit =( e ) => {
        e.preventDefault();

        setCategories(categories => [ ...categories, inputValue]);
        setInputValue('');
    };

    return (
        <form onSubmit = { handleSubmit }>

            <input
                type = "text"
                value = { inputValue }
                onChange = { handleinputChange }
            /> 

        </form>
    )
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};