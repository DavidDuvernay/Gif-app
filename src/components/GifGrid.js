import React from 'react';

export const GifGrid = ( { category }) => {
    
    const getGifs = async() => {
        const url = 'http://api.giphy.com/v1/gifs/search?q=Rick%20and%20Morty&limit=10&api_key=DjBrCGeFD2LhaQNB1Lb05osHuPubFNcg';
        
        const resp = await fetch( url );
        
        const { data } = await resp.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });

        console.log(gifs)
    };

    getGifs()

    return(
        <div> 
            <h3> { category } </h3>
        </div>
    );
    
}