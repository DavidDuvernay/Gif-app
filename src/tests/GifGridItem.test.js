import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../components/GifGridItem';

describe('Prueba de GIfGridItem para mostrar componente', () => {

    const title = 'Titulo de prueba';
    const url = 'https//localhost/imagenDePrueba.jpg';
    const wrapper = shallow( <GifGridItem title= { title } url = { url }/> );
    
    test('Debe mostrar componente', () =>{


        expect(wrapper).toMatchSnapshot();
    });

    test('Debde de tener un parrafo con title', () => {
        
        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( title );
    });

    test('Debe tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    });

    test('Debe incluid la clase animate__fadeInLeft', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes('animate__fadeInLeft') ).toBe( true );

    });
    
});