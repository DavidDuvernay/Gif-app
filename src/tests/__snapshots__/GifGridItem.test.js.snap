// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GIfGridItem para mostrar componente Debe mostrar componente 1`] = `
<div
  className="card animate__animated animate__fadeInLeft"
>
  <img
    alt="Titulo de prueba"
    src="https//localhost/imagenDePrueba.jpg"
  />
  <p>
     
    Titulo de prueba
     
  </p>
</div>
`;
